pkg/cluster/validation.go

```diff
@@ -78,3 +93,33 @@ func ValidatePartial(cluster *storage.Cluster) *errorhelpers.ErrorList {
 
 	return errorList
 }
+
+func IsNameValid(name string) error {
+	// This check is added to avoid problems with rendering of helm templates.
+	// If a number is added as a cluster name, Helm may treat is as such and
+	// convert it to scientific notation - see cluster name in sensor secret.
+	// Because the fix in the Helm chart is tricky, we reject all names that
+	// could be parsable to a number or any other Helm construct.
```

Comment:
If this comment relates to the entire `IsNameValid()`, please promote it to function's comment. If to some specific set of `if`s, move it down accordingly.